apis/metadata_validation.go

```diff
@@ -19,10 +19,21 @@ package apis
 import (
 	"fmt"
 
+	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/api/validation"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	groupName = "serving.knative.dev"
+	// CreatorAnnotation is the annotation key to describe the user that
+	// created the resource.
+	CreatorAnnotation = groupName + "/creator"
+	// UpdaterAnnotation is the annotation key to describe the user that
+	// last updated the resource.
+	UpdaterAnnotation = groupName + "/lastModifier"
```

Comment:
Aren't these constants defined somewhere already?

Reply:
yes they are defined in
https://github.com/knative/serving/blob/master/pkg/apis/serving/register.go#L20
But in knative.dev/pkg if i want to use those constants i need to import `"knative.dev/serving/pkg/apis/serving"` which leads to cyclic dependency.

Reply:
They are also defined here:
https://github.com/knative/pkg/blob/74c5d67ea097dd45bc66a6e2212ff70058efe005/webhook/user_info.go#L27-L35

We should probably move them to /apis though and reference them from both points.

Reply:
Modified

Sorry again i squashed commits next time i will take care.